File: helios/src/main/java/org/onereed/helios/common/DirectionUtil.java
package org.onereed.helios.common;

/** Static utility methods for working with angular directions. */
public class DirectionUtil {

  /**
   * Normalizes the argument angle into the range [-180..180) as a float, since that's what all the
   * Android rotation methods want.
   */
  public static float zeroCenterDeg(double deg) {
    float centered = (float) deg;

    while (centered < -180.0f) {
      centered += 360.0f;
    }

    while (centered >= 180.0f) {
      centered -= 360.0f;
    }

    return centered;
  }

  private DirectionUtil() {}
}

File: helios/src/main/java/org/onereed/helios/common/DirectionUtil.kt
package org.onereed.helios.common

import kotlin.math.IEEErem

/** Static utility methods for working with angular directions.  */
object DirectionUtil {

    /**
     * Returns the signed angular distance from [deg1] to [deg2] in the range `[-180..180]`.
     */
    @JvmStatic
    fun arc(deg1: Double, deg2: Double): Double {
        val diff = deg2 - deg1
        return diff.IEEErem(360.0)
    }
}
